hybrid/compartment_examples/inter_comp_call/secure-update_ddc/compartments-update_ddc.S

```diff
@@ -0,0 +1,53 @@
+/* Compartment from which we call the switcher to perform inter-compartment
+ * transition. The call is via a capability, to update the PCC bounds
+ * appropriately to cover `switch_compartment`.
+ */
+
+// Compartment functions
+.global comp_f_fn
+.global comp_g_fn
+
+// Labels for size computations
+.global comp_f_fn_end
+.global comp_g_fn_end
+
+.type comp_f_fn, "function"
+comp_f_fn:
+    // Set compartment ID we want to switch to
+    mov       x0, #1
+
+    // Store the `clr` for exitting `comp_f_fn`; this is overwritten by
+    // `switch_compartment`.
+    str       clr, [sp, #-16]!
+
+    // Attempt to modify DDC of `switch_compartment`
+    mrs       c1, DDC
+    ldr       c2, [x20]
```

Comment:
What about using `c20`?

Reply:
What I mean is that I think the access will work if you use `c20`, since that ignores `DDC` and just uses the capability itself (which is now leaked due to changes in `clean` for this example).

Have I misunderstood?

Reply:
What I attempted to do is store the address pointing to the DDC used by `switch_compartment` in `x20`, not the actual capability itself. `c0` in `executive_switch` (which is installed in `c20`) is the `ldpblr` capability we're using normally, so I'm just taking the address from that here.